toc/rfc/rfc-draft-service-binding-rotation.md

```diff
@@ -0,0 +1,123 @@
+
+#### GET /v3/service_credential_bindings
+
+The API `GET /v3/service_credential_bindings` will return all service credential bindings of an application including multiple bindings to the same service instance.
+
+### Mapping of service credential bindings into application containers
+
+When creating the service credential binding information like VCAP_SERVICES or file-based one the CC must select the newest one per service instance only that is in state `succeeded`. This must be based on the `created_at` and `last_operation.state` fields from the credential binding.
+
+An application restart or restage as today will be required to activate the new binding.
+
+### CF CLI
+
+Users can create multiple service credential bindings by invoking the `bind-service` command multiple times. Example:
+```
+cf bind-service myApp myService  # initial binding
+cf bind-service myApp myService  # additional binding to the same service instance
+```
+
+`cf unbind-service myApp myService` shall delete all existing service credential bindings for `myApp` to `myService`.
+An additional parameter `cf unbind-service --guid <guid>` should support the deletion of a single service credential binding.
+
+`cf service myService` should list all bindings to apps including their `guid` and `created_at` timestamp. This information is helpful to understand which binding will be mapped into the application container.
+
+The cleanup of old service credential bindings should be supported by a new CF CLI command:
+```
+cf cleanup-outdated-service-bindings myApp [--service-instance myService] [--keep-last 1]
+```
+The CLI will use the  CF API `GET /v3/service_credential_bindings?app_guids=:guid ` to list the service instance bindings for an application and should delete all old bindings based on the creation date leaving the newest service bindings. With the `keep-last` parameter, users can keep the x newest bindings per app and service instance. If no service instance name is provided, the CLI should delete the old bindings of all services currently bound to the application.
+It is in the responsibility of the user to invoke `cf cleanup-outdated-service-bindings myApp` only after a successfully restage/restart of the app, i.e. when old service credential bindings are not used anymore by any app container. 
+
+A full service binding rotation flow with the CF CLI could look like:
+```
+cf bind-service myApp myService -c {<parameters>}
+cf bind-service myApp myService2 -c {<parameters>}
+cf restage myApp --strategy rolling
+cf cleanup-outdated-service-bindings myApp
+```
+
+## Possible Future Work
+
+### CC adoption of OSBAPI 2.17
+
+The CC could use the service binding rotation functionality introduced with the OSBAPI 2.17 and introduce a dedicated API endpoint for rotation, e.g. `POST /v3/service_credential_bindings/:guid/actions/duplicate` (good naming to be found).
```

Comment:
will this naming be defined as part of this RFC?

Reply:
I don't see the need to define the naming for this in this RFC because it is future work which could happen or not.